internal/core/workers/operation_execution_worker/operation_execution_worker.go

```diff
@@ -101,18 +108,21 @@ func (w *OperationExecutionWorker) Start(ctx context.Context) error {
 			}
 
 			loopLogger.Debug("operation execution failed", zap.Error(executionErr))
+
+			onErrorStartTime := time.Now()
 			onErrorErr := executor.OnError(operationContext, op, def, executionErr)
 			if onErrorErr != nil {
 				loopLogger.Error("operation OnError failed", zap.Error(onErrorErr))
 			}
+			ReportOperationOnErrorLatency(onErrorStartTime, w.schedulerName, op.DefinitionName, onErrorErr != nil)
 		}
+		ReportOperationExecutionLatency(executeStartTime, w.schedulerName, op.DefinitionName, executionErr != nil)
```

Comment:
Shouldn't this report be called right after the `Execute` returns? Here, we're counting the `OnError` time, so in case of error, the reported time for the execution would be `ExecuteTime + OnErrorTime`.